metadata-ingestion/src/datahub/api/entities/corpgroup/corpgroup.py

```diff
@@ -39,46 +54,183 @@ class CorpGroup:
     overrideEditable: bool = False
     picture_link: Optional[str] = None
     slack: Optional[str] = None
+    admins: List[Union[str, CorpUser]] = []
+    members: List[Union[str, CorpUser]] = []
+
+    @pydantic.validator("admins", "members", each_item=True)
+    def make_urn_if_needed(v):
+        if isinstance(v, str):
+            return builder.make_user_urn(v)
+        return v
+
+    @property
+    def urn(self):
+        return builder.make_group_urn(self.id)
 
-    def __post_init__(self):
-        self.urn = builder.make_group_urn(self.id)
+    @dataclass
+    class GenerationConfig:
```

Comment:
embedded classes is kinda an anti-pattern in python - this should be at the module level

Reply:
moved up, please check if this is what you were looking for.